website/blog/2023-01-17-grouping-data-tests.md

```diff
@@ -0,0 +1,96 @@
+
+## Case Study: NYC Subway Data
+
+To demonstrate the utility (or, should I say, necessity) of group-level checks, let's consider some real-world open data from the [NYC subway system](http://web.mta.info/developers/turnstile.html) which I can always count on to have plenty of data quality quirks (which, to be clear, I do not say as a criticism. There's nothing unexpected about this in real-world "data as residue" data.). Cumulative entries through each turnstile across all subway stations are recorded 4x daily, creating a structure with one record for each turnstile and timestamp combination. 
+
+Of course, the information we want out of this data is probably not the cumulative count through some turnstile from some arbitrary start date but rather the station-level entries during a given period. So, in our transformations, we would take a lagged difference of the cumulative entries by turnstile and aggregate that up to the station-level. Just collating data from 5,000 sensors – what could go wrong, right? 
+
+However, that seemingly trivial lagged-difference transformation makes two key assumptions: the cumulative entries are <Term id="monotonically-increasing"/> _by turnstile_ and every time period observations is present _for every turnstile_.
+
+These conditions illustrate the two benefits of grouped checks we mentioned before: monotonicity can only be assessed after grouping by turnstile (there's no reason the cumulative entry count should only go up when comparing observations across different turnstiles), and although the presence of given timestamps _could_ be checked at the dataset level it is substantially more rigorous when checked at the individual sensor level. 
+
+So what do we discover when we validate our data by group?
+
+Testing for monotonicity, we find many poorly behaved turnstiles. Unlike the well-behaved dark blue line, other turnstiles seem to _decrement_ versus _increment_ with each rotation while still others cyclically increase and plummet to zero – perhaps due to maintenance events, replacements, or glitches in communication with the central server.
+
+<Lightbox src="/img/blog/2023-01-17-grouping-data-tests/1-monotonicity.png" title="Cumulative Entries by Turnstile for 3 Turnstiles" alt="A chart with three lines: one in dark blue trending up and to the right, one in light blue trending down and to the right, and one in very light blue which tracks up and then suddenly drops in a sawtooth pattern."/>
+
+CA = A033, Unit = R170, SCP = 02-00-05
+
+CA = N559, Unit = R425, SCP = 00-06-01
+
+CA = PTH03, Unit = R552, SCP = 00-00-07
```

Comment:
Not entirely sure what these are - they look weird in full size compared to the tiny font they were in the google doc